test/com/google/javascript/jscomp/ProcessCommonJSModulesTest.java

```diff
@@ -933,12 +980,70 @@ public void testIssue2616() {
             "  foo: foo,",
             "};"),
         lines(
-            "/** @const */ var module$test={};",
-            "module$test.foo = function foo() {",
+            "/** @const */ var module$test = {/** @const */ default: {}};",
+            "module$test.default.foo = function foo() {",
             "  return 1;",
             "};"));
   }
 
+  public void testFingerprintUmd() {
+    testModules(
+        "test.js",
+        lines(
+            "(function (name, context, definition) {",
+            "  'use strict';",
+            "  if (typeof define === 'function' && define.amd) {",
+            "    define(definition);",
+            "  } else if (typeof module !== 'undefined' && module.exports) {",
+            "    module.exports = definition();",
+            "  } else if (context.exports) {",
+            "    context.exports = definition();",
+            "  } else {",
+            "    context[name] = definition();",
+            "  }",
+            "})('Fingerprint2', this, function() {",
+            "  var Fingerprint2 = function() {",
+            "    if (!(this instanceof Fingerprint2)) { return new Fingerprint2(); }",
+            "  };",
+            "  return Fingerprint2$$module$test;",
+            "})"),
+        lines(
+            "/** @const */ var module$test = {};",
+            "var Fingerprint2$$module$test = function() {",
+            "  if (!(this instanceof Fingerprint2$$module$test)) {",
```

Comment:
From @blickly 
I think this should be Fingerprint2

Reply:
I'm not following ... The rewriting looks correct.

Reply:
Oh I think this was meant for line 1008